internal/settings/ovh.go

```diff
@@ -142,3 +169,77 @@ func (o *ovh) Update(ctx context.Context, client *http.Client, ip net.IP) (newIP
 		return nil, fmt.Errorf("%w: %s", ErrUnknownResponse, s)
 	}
 }
+
+func (o *ovh) updateWithZoneDNS(client *ovhClient.Client, ip net.IP) (newIP net.IP, err error) {
+	recordType := A
+	var ipStr string
+	if ip.To4() == nil { // IPv6
+		recordType = AAAA
+		ipStr = ip.To16().String()
+	} else {
+		ipStr = ip.To4().String()
+	}
+	// subDomain filter of the ovh api expect an empty string to get @ record
+	subDomain := o.host
+	if subDomain == "@" {
+		subDomain = ""
+	}
+	// get existing records
+	var recordIDs []uint64
+	if err := client.Get(fmt.Sprintf("/domain/zone/%s/record?fieldType=%s&subDomain=%s", o.domain,
+		recordType, subDomain), &recordIDs); err != nil {
+		return ip, err
```

Comment:
```suggestion
		return nil, err
```

just for clarity, the newIP is `nil` because we encountered an error. The comment applies for the few returns below.